swift/Arrow/Sources/Arrow/ArrowArrayBuilder.swift

```diff
@@ -118,6 +118,56 @@ public class Time64ArrayBuilder: ArrowArrayBuilder<FixedBufferBuilder<Time64>, T
     }
 }
 
+public class StructArrayBuilder: ArrowArrayBuilder<StructBufferBuilder, StructArray> {
+    let builders: [any ArrowArrayHolderBuilder]
+    let fields: [ArrowField]
+    public init(_ fields: [ArrowField], builders: [any ArrowArrayHolderBuilder]) throws {
```

Comment:
Ah, we can use `throws` for `init()`.
We may want to add `throws` to `ArrowArray.init()` and remove `StructArrowArray.initialize()`. But it breaks compatibility...

Reply:
Agreed, Adding the throw to init would break compatibility.  I added the initialize method to the ArrowArray base class that way it will/should always be called for all ArrowArray.  Let me know what you think.

Reply:
If we need to call `initialize()` for all `ArrowArray`, I like the `init() throws` approach. Because users can't forget to call `initialize()` with the `init() throws` approach.

In general, do Swift users call `ArrowArray.init()` directory? Or they use builder API to build an array?

Reply:
Gotcha, the builders are the only place that should be calling init but since the method is public it can be called by anyone.  I will update the init method in ArrowArray to throw.  Is there a way to indicate in the PR or issue that this is a breaking change?

Reply:
Could you add `**This PR contains a "Critical Fix".**` to the PR description?
See also our PR template: https://raw.githubusercontent.com/apache/arrow/main/.github/pull_request_template.md